app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainFragment.kt

```diff
@@ -106,17 +104,16 @@ class MainFragment : Fragment() {
 
     fun updateAddressUI(location: Location) {
 
-        var geocoder: Geocoder
-        var addressList = ArrayList<Address>()
+        val geocoder: Geocoder
 
         geocoder = Geocoder(activity!!.applicationContext, Locale.getDefault())
 
-        addressList = geocoder.getFromLocation(
+        var addressList = geocoder.getFromLocation(
                 location.latitude,
                 location.longitude,
                 1
         ) as ArrayList<Address>
-        enterCityname.setText(addressList.get(0).getAddressLine(0))
+        enterCityname.setText(addressList[0].getAddressLine(0))
```

Comment:
Another good thing to do here: check the size of the array.  Make sure it has at least one item in it.  Otherwise, you could get ArrayIndexOutOfBoundsException.

Reply:
After Our team meeting we notice that and will end up having to  check the array for only the city, so this is one thing we found out we needed to look into after finding it aswell, good catch.